pkg/cloudinit/scripts/configure-snapstore-proxy.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -xe
+
+# Assumptions:
+#   - snapd is installed
+#   - /capi/etc/snapstore-proxy-scheme contains the snapstore scheme
+#   - /capi/etc/snapstore-proxy-domain contains the snapstore domain
+#   - /capi/etc/snapstore-proxy-id contains the snapstore id
+
+SNAPSTORE_PROXY_SCHEME=$(cat /capi/etc/snapstore-proxy-scheme)
+SNAPSTORE_PROXY_DOMAIN=$(cat /capi/etc/snapstore-proxy-domain)
+SNAPSTORE_PROXY_ID=$(cat /capi/etc/snapstore-proxy-id)
+
+if [ -z "${SNAPSTORE_PROXY_SCHEME}" ] || [ -z "${SNAPSTORE_PROXY_DOMAIN}" ] || [ -z "${SNAPSTORE_PROXY_ID}" ]; then
```

Comment:
Any reason to not check for file existence? `cat` will fail on a non existent path, how does this interact with `-z` ?

Reply:
If one of the files don't exist then we'll never reach the if. This is because of `set -e`. I'll think of a better solution.